src/main/java/net/datafaker/datafaker_gen/sink/TextFileSink.java

```
@@ -15,7 +18,23 @@ public String getName() {
 
     @Override
     public void run(Map<String, String> config, Function<Integer, ?> function, int numberOfLines) {
-        try (BufferedWriter bw = Files.newBufferedWriter(Path.of(config.get("filename")))) {
+        String filepath = config.get("filepath");
+        String mode = config.get("mode");
+        Path path = Paths.get(filepath);
```

In that case probably need to check that `path`: `!Files.isDirectory(path) && Files.isWritable(path)`

---

`!Files.isDirectory(path)` - makes sense.
Probably this `Files.isWritable(path)`  will make sense in case if write mode is append, because it will return `false` if file doesn't exist.

---

yep, agree

---

Should we throw an exception in case if `Files.isDirectory(path)` - is `true` ? Or create a file with some default name?

---

1. better throw exception

+ 2. there is one more case
since there is a way to specify a path then it's possible to  be in folder `/usr/local/tests` specify path `a/b/c/d/output` where `a`, `b`, `c`, `d` folders are not exist. Probably need to have another option: create tree of folders if not present or not

---

Easier to say, if path doesn't exist - we have to create it, right?

---

Something like `fail_if_absent_dir` ?

---

usually it is called creation of parent dirs
so may be something like that `create_parent_dirs`

---

Done. Added `create_parent_dirs` config.